### DISCUSSION TITLE
TreeView

### DISCUSSION THREAD

**Discussion by MrHause** [Daguerreo/QMLTreeView#11] [General]
Hello,
I have the question for you about the treeView. I see that there is no TreeView in QtQuick.Controls 2. Is this TreeView implemented by you? Did you have some source inspiration how to write it? I am trying to figure it out, how is it possible that such basic thing like TreeView is missing in new QtQuick. I am new in Qt and so far  I am too bad to write custom modules like this TreeView. So I look for some information and good sources how to implement such modules. 

This project looks really good. Good job.

Best regards

**Comment by Daguerreo** [comment]
Hello,

A little story about Tree View and Qt Quick. At the beginning it wasn't part of the QtQuick controls set, later when the marketplace was added, Qt Company implemented it as a payed add-on. However the usage was overkill for mostly of the simple uses and many people thought was ridicolous to provide a payed add-on so many provided a custom implementation, including me.

There was a good change of mind though, from Qt 6.3 they added a standard component with a TreeView, so if you go with Qt 6 you can just go with that right now, this plugin is meant for Qt 5 and if you don't actually need the capabilities of the add-on component.

Finally, this is a pure QML implementation, so it's not built to work with huge quantity of entries and it's not optimized for that because it would need a C++ implementation instead. I wrote this because I needed a easy to use general purpose tree view, not a all-fit-in-one solution.

**Comment by MrHause** [comment]
This is interesting what you write. It is hard for new users in qt because we are missing some information and I have more questions than answers. I started my qt project with Qt6 but then I switched to Qt5 because so far I did not find any example for qt6 how to use TreeView in QML that is based on c++ model. I found some example for Qt5 but they are not compatible for qt6. Still my knowledge is not enough to make such application only based on documentation. But work is still in progress. Your answer helped me and I appreciate it.

Update:
Thanks to your suggestion I was able to run treeView in qml using qt6.

**Reply by Daguerreo** [reply]
You can find a lot of example inside Qt Creator in the project example, documentation it's great though. Then it's up to you to have a bit of patience learning everything you need because the framework it's actually huge, but there are a lot of resources out there.
